executable-spec/src/Cardano/Ledger/Generators/QuickCheck.hs

```diff
@@ -80,17 +82,31 @@ boundedWith aMinBound aMaxBound maxDifference =
   where mid = aMinBound + (aMaxBound - aMinBound) `div` 2
 
 prvNoQuorum :: Gen Word8
-prvNoQuorum = Gen.choose (3, 7)
+prvNoQuorum =  Gen.choose (1, 3)
 
 prvNoMajority :: Gen Word8
-prvNoMajority = Gen.choose (3, 7)
+prvNoMajority = Gen.choose (1, 3)
 
 rA :: Gen Float
 rA = Gen.choose (0, 0.5)
 
-stakeDist :: Gen (StakeDistribution p)
-stakeDist = do
-  someParticipants <- participants
+stakeDist :: (Participants p) -> Gen (StakeDistribution p)
+stakeDist someParticipants = do
+  Gen.oneof [ stakeDistUniform someParticipants
+            , stakeDistSkewed someParticipants
+            ]
+
+stakeDistUniform :: (Participants p) -> Gen (StakeDistribution p)
+stakeDistUniform someParticipants = do
   let vkeys = Set.toList $ dom someParticipants
   stks <- Gen.vectorOf (length vkeys) (Gen.choose (1, 20))
```

Comment:
Don't we want to test what happens if the stake of a key is `0`?

Reply:
In principle, in a PoS system, a participant must own some stake, otherwise one cannot be a participant.

Reply:
Done OK. I have tested it with 0 stake and have allowed this case to enter the test cases.

Reply:
I also asked Jared whether a zero stake holder is allowed.

Reply:
For the record, a zero stake is allowed in Shelley.